src/engine/runtime.js

```diff
@@ -497,6 +505,41 @@ class Runtime extends EventEmitter {
         this.emit(Runtime.EXTENSION_ADDED, categoryInfo.blocks.concat(categoryInfo.menus));
     }
 
+    /**
+     * Reregister the primitives for an extension
+     * @param  {ExtensionInfo} extensionInfo - new info (results of running getInfo)
+     *                                         for an extension
+     * @private
+     */
+    _refreshExtensionPrimitives (extensionInfo) {
+        let extensionBlocks = [];
+        for (const categoryInfo of this._blockInfo) {
+            if (extensionInfo.id === categoryInfo.id) {
```

Comment:
Now this gets called for each extension instead of for the whole list. Does the VM know which index in _blockInfo corresponds to a particular extension as that would save having to look through the whole array?

Reply:
The VM doesn't currently index blocks by category or extension ID. It could in the future if there's enough benefit, though.

I think this is fine for now, but if you, say, don't like the extra indent you could use `if (extensionInfo.id === categoryInfo.id) continue;`, or if you want to be fancy you could do `for (const categoryInfo of this._blockInfo.filter(x => extensionInfo.id === x.categoryInfo.id) {`. Like I said, though, I think this is fine as it is :)